storage/watchable_store.go

```diff
@@ -356,3 +367,24 @@ type watching struct {
 	// The chan might be shared with other watchings.
 	ch chan<- storagepb.Event
 }
+
+// unsafeAddWatching puts watching with key k into watchableStore's synced.
+// Make sure to this is thread-safe using mutex before and after.
+func unsafeAddWatching(synced *map[string]map[*watching]struct{}, k string, wa *watching) error {
+	mp := *synced
+	if v, ok := mp[k]; ok {
+		if wa != nil {
```

Comment:
Agree. I will return error for `nil` watching. Or should I just assume it't not `nil`?
It's useful for some testing when you don't want to create dummy watching. Your call. Thanks,